source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACOAdaptiveCard.mm

```diff
@@ -29,8 +31,17 @@ + (ACOAdaptiveCardParseResult *)fromJson:(NSString *)payload;
             result.card = card;
             result.IsValid = YES;
         }
-        catch(...)
+        catch(const AdaptiveCardParseException& e)
         {
+            // converts AdaptiveCardParseException to NSError
+            ErrorStatusCode errorStatusCode = e.GetStatusCode();
+            NSInteger errorCode = (long)errorStatusCode;
+            NSString *errorMessage= [NSString stringWithCString:e.GetMessage().c_str()
+                                                  encoding:[NSString defaultCStringEncoding]];
+            NSError *parseError = [NSError errorWithDomain:ACRParseErrorDomain
+                                                      code:errorCode
+                                                  userInfo:@{NSLocalizedFailureReasonErrorKey:errorMessage}];
```

Comment:
>NSLocalizedFailureReasonErrorKey:errorMessage [](start = 61, length = 45)

This error message isn't actually going to be localized.

Reply:
The error string will still not be localized. Just calling NSLocalizedString doesn't localize the string for you.

Assuming that the Adaptive Cards team is not planning to localize its error messages any time soon, I would recommend not putting the error message into the localizedFailureReason property.

---
In reply to: [161899565](https://github.com/Microsoft/AdaptiveCards/pull/1074#discussion_r161899565) [](ancestors = 161899565)

Reply:
Understood, thanks Ben

---
In reply to: [161927305](https://github.com/Microsoft/AdaptiveCards/pull/1074#discussion_r161927305) [](ancestors = 161927305,161899565)